base/src/main/java/io/spine/base/Environment.java

```diff
@@ -21,37 +21,71 @@
 package io.spine.base;
 
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import io.spine.annotation.Internal;
 import io.spine.annotation.SPI;
 import org.checkerframework.checker.nullness.qual.Nullable;
 
+import static com.google.common.base.Preconditions.checkState;
+import static io.spine.base.BaseEnvironmentType.PRODUCTION;
+import static io.spine.base.BaseEnvironmentType.TESTS;
+
 /**
  * Provides information about the environment (current platform used, etc.).
+ *
+ * <p><b>When extending, please note</b> that this class does not handle the situations when two
```

Comment:
Let's rephrase in terms of the responsibitliy of the framework user. It's not that this class handles or does not handle any use cases. It just delegates the responsibility to tell between different `PRODUCTION` environments to the developer writing the code.